Makefile

```diff
@@ -36,3 +36,31 @@ parsers/tree-sitter-c-sharp.wasm: node_modules/tree-sitter-c-sharp/package.json
 	mkdir -p $(dir $@)
 	npx tree-sitter build-wasm $(dir $^)
 	mv tree-sitter-c_sharp.wasm $@
+
+
+# Build web-tree-sitter
+
+WEB_TREE_SITTER_FILES := README.md package.json tree-sitter-web.d.ts tree-sitter.js tree-sitter.wasm
+WEB_TREE_SITTER_DIR := vendor/web-tree-sitter
+
+MAKE_CACHE_DIR := .make-work
+
+.PHONY: web-tree-sitter
+web-tree-sitter: $(addprefix $(WEB_TREE_SITTER_DIR)/,$(WEB_TREE_SITTER_FILES))
+
+$(addprefix $(WEB_TREE_SITTER_DIR)/,$(WEB_TREE_SITTER_FILES)): tree-sitter-version
+	@rm -rf $(MAKE_CACHE_DIR)/tree-sitter
+	@TREE_SITTER_VERSION=$(shell cat tree-sitter-version) ;\
+	git clone                                       \
+		-c advice.detachedHead=false --quiet           \
+		--depth=1 \
+		https://github.com/tree-sitter/tree-sitter.git \
+		$(MAKE_CACHE_DIR)/tree-sitter ;\
+	cd $(MAKE_CACHE_DIR)/tree-sitter && \
+	git fetch origin $$TREE_SITTER_VERSION && \
```

Comment:
Do we need a `git clone`? Can we fetch a tarball archive instead?

Reply:
done!